### PR TITLE
GAWB-1276: Allow thurloe to send notifications to external emails

### PR DESCRIPTION
**Pull request by MatthewBemis**
This gives orchestration a way to specify an email directly and bypass the preferredEmail settings that won't exist for unregistered users.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [x] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/thurloe/dataaccess/SendGridDAO.scala [hunk 1]
```diff
@@ -21,8 +21,11 @@ trait SendGridDAO {
 
   def sendNotifications(notifications: List[Notification]): Future[List[Response]] = {
     Future.sequence(notifications.map { notification =>
-      lookupPreferredEmail(notification.userId) flatMap { preferredEmail =>
-        val email = createEmail(preferredEmail, notification.replyTo, notification.notificationId, notification.substitutions)
+      val toAddressFuture = notification.userId map (lookupPreferredEmail) getOrElse
+        Future.successful(notification.userEmail.getOrElse(throw new NotificationException("No recipient specified", Seq.empty, notification.notificationId)))
```
Comment (dvoet) [inline src/main/scala/thurloe/dataaccess/SendGridDAO.scala]: seems like you should use the email sent in the request over the preferred email, this does the opposite

**Review by dvoet — CHANGES_REQUESTED** [review]

File: src/main/resources/swagger/thurloe.yaml [hunk 2]
```diff
@@ -324,11 +324,13 @@ paths:
         schema:
           {
             "type": "object",
-            "required": ["userId", "notificationId"],
```
Comment (breilly2) [inline src/main/resources/swagger/thurloe.yaml:327]: The `Notification` case class looks like it still expects to always have a `notificationId`. Should it still be required?

Comment (MatthewBemis) [reply src/main/resources/swagger/thurloe.yaml:327]: Yes, dropped this by mistake

**Comment by breilly2** [comment]
👍 with update to make `notificationId` required again in swagger

**Review by dvoet — APPROVED** [review]
👍

**Pull request merged**